### PR TITLE
allow to customize healthz & webhook-server port

### LINKED ISSUE #2411: Customize healthz and webhook server port on addons manager package
(This is used to request new product features)

**Describe the feature request**
1. Make the health port configurable in addons-manager package
2. Make the webhook server port configurable in addons-manager package
3. Make the same changes in framework and tkg composite package.

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [X] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by ggpaue**
### What this PR does / why we need it

This PR is to allow customize healthz & webhook-server port in addons-manager

### Which issue(s) this PR fixes

Fixes #2411 

### Describe testing done for PR

manually build addons-manager package and deploy to tanzu cluster to see if it can be reconciled successfully

### Release note

```release-note
package-based-lcm: Made healthz & webhook server port configurable in addons-manager package
```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: packages/management/addons-manager/bundle/config/upstream/addons-manager.yaml [hunk 1]
```diff
@@ -261,10 +261,10 @@ spec:
             cpu: 100m
             memory: 40Mi
         ports:
-          - containerPort: 9865
+          - containerPort: #@ data.values.tanzuAddonsManager.webhookServerPort
             name: webhook-server
             protocol: TCP
-          - containerPort: 18316
+          - containerPort: #@ data.values.tanzuAddonsManager.healthzPort
```
Comment (vijaykatam) [inline packages/management/addons-manager/bundle/config/upstream/addons-manager.yaml:267]: Need to update deployment with this datavalue for healthz port.

Comment (ggpaue) [reply packages/management/addons-manager/bundle/config/upstream/addons-manager.yaml:267]: updated the healthz port with data value

File: packages/management/addons-manager/bundle/config/upstream/addons-manager.yaml [hunk 2]
```diff
@@ -246,7 +246,7 @@ spec:
       - args:
         - --metrics-bind-addr=0
         - --leader-elect=false
-        - --health-addr=:18316
+        - #@ "--health-addr=:{}".format(data.values.tanzuAddonsManager.deployment.healthzPort)
         - #@ "--addon-namespace={}".format(data.values.tanzuAddonsManager.namespace)
         #@ if/end data.values.tanzuAddonsManager.featureGates.clusterBootstrapController:
         - --feature-gate-cluster-bootstrap=true
```
Comment (vijaykatam) [inline packages/management/addons-manager/bundle/config/upstream/addons-manager.yaml:252]: Where is the param `webhook-server-port` being passed?

File: addons/main.go [hunk 3]
```diff
@@ -130,6 +130,9 @@ func parseAddonFlags(addonFlags *addonFlags) {
 }
 
 func main() {
+	var webhookServerPort int
+	flag.IntVar(&webhookServerPort, "webhook-server-port", 9865, "The port that the webhook server serves at.")
```
Comment (shyaamsn) [inline addons/main.go:134]: The other flags are defined at the top. Could we follow the same convention as other flags.

File: packages/management/addons-manager/bundle/config/values.yaml [hunk 4]
```diff
@@ -10,6 +10,7 @@ tanzuAddonsManager:
     priorityClassName: null
     nodeSelector: null
     tolerations: []
+    webhookServerPort: 9865
```
Comment (shyaamsn) [inline packages/management/addons-manager/bundle/config/values.yaml:13]: We also need to make the same changes in framework and tkg composite package.

File: packages/management/addons-manager/bundle/config/upstream/addons-manager.yaml (hunk 2 above)
Comment (ggpaue) [reply packages/management/addons-manager/bundle/config/upstream/addons-manager.yaml:252]: @vijaykatam which one should we use?
`webhook-server-port=:9865` or `webhook-server-port=9865`

Comment (vijaykatam) [reply packages/management/addons-manager/bundle/config/upstream/addons-manager.yaml:252]: `- #@ "--webhook-server-port=:{}".format(data.values.tanzuAddonsManager.deployment.webhookServerPort)`

Comment (ggpaue) [reply packages/management/addons-manager/bundle/config/upstream/addons-manager.yaml:253]: added webhook-server-port in the args

File: packages/management/addons-manager/bundle/config/values.yaml (hunk 4 above)
Comment (ggpaue) [reply packages/management/addons-manager/bundle/config/values.yaml:13]: added these data values in framework and tkg composite package as well

**Review by vijaykatam — APPROVED** [review]

**Pull request merged**